ext/MooncakeSpecialFunctionsExt.jl

```diff
@@ -3,44 +3,44 @@ module MooncakeSpecialFunctionsExt
 using SpecialFunctions, Mooncake
 using Base: IEEEFloat
 
-import Mooncake: @from_rrule, DefaultCtx, @zero_adjoint
+import Mooncake: DefaultCtx, @from_chain_rule, @zero_derivative
 
-@from_rrule DefaultCtx Tuple{typeof(airyai),IEEEFloat}
-@from_rrule DefaultCtx Tuple{typeof(airyaix),IEEEFloat}
-@from_rrule DefaultCtx Tuple{typeof(airyaiprime),IEEEFloat}
-@from_rrule DefaultCtx Tuple{typeof(airybi),IEEEFloat}
-@from_rrule DefaultCtx Tuple{typeof(airybiprime),IEEEFloat}
-@from_rrule DefaultCtx Tuple{typeof(besselj0),IEEEFloat}
-@from_rrule DefaultCtx Tuple{typeof(besselj1),IEEEFloat}
-@from_rrule DefaultCtx Tuple{typeof(bessely0),IEEEFloat}
-@from_rrule DefaultCtx Tuple{typeof(bessely1),IEEEFloat}
-@from_rrule DefaultCtx Tuple{typeof(dawson),IEEEFloat}
-@from_rrule DefaultCtx Tuple{typeof(digamma),IEEEFloat}
-@from_rrule DefaultCtx Tuple{typeof(erf),IEEEFloat}
-@from_rrule DefaultCtx Tuple{typeof(erf),IEEEFloat,IEEEFloat}
-@from_rrule DefaultCtx Tuple{typeof(erfc),IEEEFloat}
-@from_rrule DefaultCtx Tuple{typeof(logerfc),IEEEFloat}
-@from_rrule DefaultCtx Tuple{typeof(erfcinv),IEEEFloat}
-@from_rrule DefaultCtx Tuple{typeof(erfcx),IEEEFloat}
-@from_rrule DefaultCtx Tuple{typeof(logerfcx),IEEEFloat}
-@from_rrule DefaultCtx Tuple{typeof(erfi),IEEEFloat}
-@from_rrule DefaultCtx Tuple{typeof(erfinv),IEEEFloat}
-@from_rrule DefaultCtx Tuple{typeof(gamma),IEEEFloat}
-@from_rrule DefaultCtx Tuple{typeof(invdigamma),IEEEFloat}
-@from_rrule DefaultCtx Tuple{typeof(trigamma),IEEEFloat}
-@from_rrule DefaultCtx Tuple{typeof(polygamma),Integer,IEEEFloat}
-@from_rrule DefaultCtx Tuple{typeof(beta),IEEEFloat,IEEEFloat}
-@from_rrule DefaultCtx Tuple{typeof(logbeta),IEEEFloat,IEEEFloat}
-@from_rrule DefaultCtx Tuple{typeof(logabsgamma),IEEEFloat}
-@from_rrule DefaultCtx Tuple{typeof(loggamma),IEEEFloat}
-@from_rrule DefaultCtx Tuple{typeof(expint),IEEEFloat}
-@from_rrule DefaultCtx Tuple{typeof(expintx),IEEEFloat}
-@from_rrule DefaultCtx Tuple{typeof(expinti),IEEEFloat}
-@from_rrule DefaultCtx Tuple{typeof(sinint),IEEEFloat}
-@from_rrule DefaultCtx Tuple{typeof(cosint),IEEEFloat}
-@from_rrule DefaultCtx Tuple{typeof(ellipk),IEEEFloat}
-@from_rrule DefaultCtx Tuple{typeof(ellipe),IEEEFloat}
+@from_chain_rule DefaultCtx Tuple{typeof(airyai),IEEEFloat}
```

Comment:
A relatively minor comment:  `from_chainrules` is more precise than `from_chain_rule`. The former clarifies that we are importing a rule from `ChainRules`, while the latter mislead me since I thought it refers to the generic chain rule terminology. 

```suggestion
@from_chainrules DefaultCtx Tuple{typeof(airyai),IEEEFloat}
```

Reply:
Ah, interesting. I have no strong view either way, so I'm happy to change it if you think it's `from_chainerules` is clearer.

Reply:
I think the name `from_chainerules` is more consistent with other use cases of `from_*` and, therefore, easier to read. 

Also, I have noticed several packages using `@from_rule` to import `ChainRules` rules, so we need to deprecate this instead of simply renaming it.